src/frontend/components/UI/AuthModal/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useRef } from 'react'
+import styles from './index.module.scss'
+import { ModalAnimation } from '@hyperplay/ui'
+import { WebviewTag } from 'electron'
+import { observer } from 'mobx-react-lite'
+import authModalState from '../../../state/authModalState'
+import { ethers } from 'ethers'
+import extensionStore from '../../../store/ExtensionStore'
+import onboardingStore from '../../../store/OnboardingStore'
+
+// TODO: replace this with dev portal prod URL when it's ready
+const url =
+  'https://hyperplay-dev-git-feature-unified-auth-ui-valist.vercel.app/signin'
+
+const METAMASK_ALREADY_PROVIDED_ERROR_CODE = -32002
+
+const isTooManyRequestsError = (error: string) => {
+  return error.includes(METAMASK_ALREADY_PROVIDED_ERROR_CODE.toString())
+}
+
+const AuthModal = () => {
+  const webviewRef = useRef<WebviewTag>(null)
+
+  const handleAccountNotConnected = async () => {
+    const currentProvider = await window.api.getCurrentWeb3Provider()
+
+    if (!currentProvider) {
+      onboardingStore.openOnboarding()
+      return
+    }
+
+    // try to trigger metamask popup to connect account
+    try {
+      const provider = new ethers.BrowserProvider(window.ethereum)
+      await provider.getSigner()
+    } catch (e) {
+      // if it fails, open the popup manually
+      // since there are already requests in the queue, this will resume
+      // the connection flow after the user unlocks metamask
+      if (isTooManyRequestsError(String(e))) {
+        extensionStore.setIsPopupOpen(true)
+      }
+    }
+  }
+
+  useEffect(() => {
+    const webview = webviewRef.current
+    if (!webview) return
+
+    const handleIpcMessage = async (event: Electron.IpcMessageEvent) => {
+      switch (event.channel) {
+        case 'closeAuthModal':
+          authModalState.closeModal()
+          break
+        case 'auth:accountConnected':
+          alert('account connected')
+          authModalState.closeModal()
+          break
+        case 'auth:accountNotConnected':
+          // TODO: try to resume flow after connecting account
```

Comment:
did you want to handle that on this branch?

Reply:
I was planning to do it in a different branch but with your suggestions, I may have an idea of how to do it.